onnxruntime/core/providers/cpu/containers.h

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#if defined(USE_MIMALLOC_STL_ALLOCATOR)
+#include <mimalloc.h>
+#endif
+
+namespace onnxruntime {
+
+template <class T>
+struct BFCArenaAllocator {
+  AllocatorPtr arena;
+  typedef T value_type;
+
+  using propagate_on_container_copy_assignment = std::true_type;
+  using propagate_on_container_move_assignment = std::true_type;
+  using propagate_on_container_swap = std::true_type;
+  using is_always_equal = std::true_type;
+
+  BFCArenaAllocator(AllocatorPtr a) noexcept {
```

Comment:
nit: AllocatorPtr is a shared_ptr. Do you really want to increment it's refcount by doing this copy only to decrement it immediately at the end of this function?